Journal/Journal/Core Data/EntryController.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  EntryController.swift
+//  Journal
+//
+//  Created by Mitchell Budge on 6/3/19.
+//  Copyright © 2019 Mitchell Budge. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+class EntryController {
+    
+    init() {
+        fetchEntriesFromServer()
+    }
+    
+    //MARK: - Methods
+    
+    func saveToPersistentStore() {
+        let moc = CoreDataStack.shared.mainContext
+        do {
+            try moc.save()
+        } catch {
+            NSLog("Error saving managed object context: \(error)")
+        }
+    }
+    
+    func createEntry(title: String, bodyText: String, timestamp: Date, identifier: String, mood: Mood) {
+        let entry = Entry(title: title, bodyText: bodyText, timestamp: timestamp, identifier: identifier, mood: mood)
+        put(entry: entry)
+        saveToPersistentStore()
+    }
+    
+    func updateEntry(entry: Entry, title: String, bodyText: String, mood: String) {
+        entry.title = title
+        entry.bodyText = bodyText
+        entry.mood = mood
+        put(entry: entry)
+        saveToPersistentStore()
+    }
+    
+    func deleteEntry(entry: Entry) {
+        let moc = CoreDataStack.shared.mainContext
+        moc.delete(entry)
+        deleteEntryFromServer(entry: entry)
+        saveToPersistentStore()
+    }
+
+    // MARK: - Networking
+    
+    typealias CompletionHandler = (Error?) -> Void
+    
+    let baseURL = URL(string: "https://journal-72b8d.firebaseio.com/")!
+    
+    func fetchEntriesFromServer(completion: @escaping CompletionHandler = { _ in}) {
+
+        let requestURL = baseURL.appendingPathExtension("json")
+        
+        var request = URLRequest(url: requestURL)
+        request.httpMethod = "GET"
+        
+        URLSession.shared.dataTask(with: requestURL) { (data, _, error) in
+            if let error = error {
+                NSLog("Error fetching entries: \(error)")
+                completion(error)
+                return
+            }
+
+            guard let data = data else {
+                NSLog("No data returned by data task")
+                completion(NSError())
+                return
+            }
+                do {
+                    let entryRepresentations = Array(try JSONDecoder().decode([String: EntryRepresentation].self, from: data).values)
+                    let moc = CoreDataStack.shared.container.newBackgroundContext()
+                    try self.updateEntries(with: entryRepresentations, context: moc)
+                } catch {
+                    NSLog("Error decoding entry representations: \(error)")
+                    completion(error)
+                    return
+                }
```

Comment:
Before resume, please pass in `completion(nil)` and this will fix your pull to refresh.